crates/linker/src/lib.rs

```diff
@@ -136,10 +136,62 @@ fn generate_dynamic_lib(
         }
     }
 
-    let bytes = crate::generate_dylib::generate(target, &custom_names)
-        .unwrap_or_else(|e| internal_error!("{}", e));
+    if !dynhost_is_up_to_date(target, dummy_lib_path, &custom_names) {
+        let bytes = crate::generate_dylib::generate(target, &custom_names)
+            .unwrap_or_else(|e| internal_error!("{}", e));
+
+        std::fs::write(dummy_lib_path, &bytes).unwrap_or_else(|e| internal_error!("{}", e))
+    }
+}
+
+fn object_matches_target<'a>(target: &Triple, object: &object::File<'a, &'a [u8]>) -> bool {
+    use target_lexicon::{Architecture as TLA, OperatingSystem as TLO};
+
+    match target.architecture {
+        TLA::X86_64 => {
+            if object.architecture() != object::Architecture::X86_64 {
+                return false;
+            }
+
+            let actual = object.format();
+
+            let expected = match target.operating_system {
+                TLO::Linux => object::BinaryFormat::Elf,
+                other => internal_error!("unexpected target {:?}", other),
+            };
+
+            actual == expected
+        }
+        TLA::Aarch64(_) => object.architecture() == object::Architecture::Aarch64,
+        _ => false,
+    }
+}
+
+fn dynhost_is_up_to_date(target: &Triple, dummy_lib_path: &Path, custom_names: &[String]) -> bool {
```

Comment:
Does this say something about dynhost, or just the dummy lib? Just a bit confused by the naming?

Reply:
yes sorry mixed those up. made it clearer now